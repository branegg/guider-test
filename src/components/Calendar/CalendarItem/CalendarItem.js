import React from 'react';
import classNames from 'classnames';

const CalendarItem = ({ dayObject, isAvailable, onClick, isActive }) => {
  const { day, weekdayLong, weekdayShort } = dayObject;
  const dayLength = day.toString().length;
  const lastDayChar = day.toString()[dayLength - 1];
  let dateSufix = 'th';

  switch (lastDayChar) {
    case '1':
      dateSufix = 'st';
      break;
    case '2':
      dateSufix = 'nd';
      break;
    case '3':
      dateSufix = 'rd';
      break;
    default:
      break;
  }

  return (
    <div
      label={weekdayLong}
      data-testid='calendar-item'
      className={classNames({
        Calendar__item: true,
        'Calendar__item--available': isAvailable,
        'Calendar__item--active': isActive,
      })}
      onClick={() => onClick()}
    >
      <p className='Calendar__date'>
        {day}
        <span data-testid='date-sufix' className='Calendar__date--small'>
          {dateSufix}
        </span>
      </p>
      <p data-testid='calendar-day' className='Calendar__day'>
        {weekdayShort}
      </p>
    </div>
  );
};

export default CalendarItem;
