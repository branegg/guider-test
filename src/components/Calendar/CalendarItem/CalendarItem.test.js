import React from 'react';
import CalendarItem from './CalendarItem';
import { render, cleanup } from '@testing-library/react';
import { DateTime } from 'luxon';
import renderer from 'react-test-renderer';

afterEach(cleanup);

const props = {
  dayObject: DateTime.local(),
  isAvailable: true,
  onClick: jest.fn(),
  isActive: true,
};

describe('CalendarItem', () => {
  it('renders without crashing', () => {
    render(<CalendarItem {...props} />);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<CalendarItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
