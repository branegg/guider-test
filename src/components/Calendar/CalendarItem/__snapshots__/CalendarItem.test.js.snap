// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarItem matches snapshot 1`] = `
<div
  className="Calendar__item"
  data-testid="calendar-item"
  label="Wednesday"
  onClick={[Function]}
>
  <p
    className="Calendar__date"
  >
    8
    <span
      className="Calendar__date--small"
      data-testid="date-sufix"
    >
      th
    </span>
  </p>
  <p
    className="Calendar__day"
    data-testid="calendar-day"
  >
    Wed
  </p>
</div>
`;
